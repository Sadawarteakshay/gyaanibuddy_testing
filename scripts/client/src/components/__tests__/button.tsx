import React from 'react';

import { render, cleanup, screen, fireEvent, waitFor } from '../../test-utils';
import Button from '../button';

import renderer from 'react-test-renderer';
import { colors } from '../../styles';

describe('Button', () => {
  // automatically unmount and cleanup DOM after the test is finished.
  afterEach(cleanup);

  it('renders without error', () => {
    const HelloWorldButton = <Button>Hello World</Button>;

    render(HelloWorldButton);

    // Snapshot test - ensure button has no unsuspecting changes between commits
    const ButtonTree = renderer
      .create(HelloWorldButton)
      .toJSON();

    expect(ButtonTree).toMatchSnapshot();
  });

  // it('button background color should change color when hovered over', async () => {
  //   const HelloWorldButton = <Button>Hello World</Button>;

  //   render(HelloWorldButton);

  //   fireEvent.mouseOver(screen.getByText('Hello World'));

  //   expect(screen.)
  // });
});
