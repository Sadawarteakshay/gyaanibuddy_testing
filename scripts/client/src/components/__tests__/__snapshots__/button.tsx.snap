// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders without error 1`] = `
<button
  className="css-wwcn44"
>
  Hello World
</button>
`;
